### ISSUE TITLE
Plugin dev: editing HTML files during `postBuild` not working properly

### ISSUE THREAD

**Issue by 0xCaso** [facebook/docusaurus#8832]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#issues).

### Prerequisites

- [X] I'm using the latest version of Docusaurus.
- [X] I have tried the `npm run clear` or `yarn clear` command.
- [X] I have tried `rm -rf node_modules yarn.lock package-lock.json` and re-installing packages.
- [X] I have tried creating a repro with https://new.docusaurus.io.
- [X] I have read the console error message carefully (if applicable).

### Description

I'm writing a custom plugin to edit the SVGs contained in the HTML file created with the build, so I'm using the `postBuild` lifecycle to do so.

For example, I want to delete all the SVGs with a certain class. It appears that, if I run `npm run serve`, docusaurus still renders the SVGs, also if they've been deleted from the HTML files in the `build` folder.

I've noticed that if I clean the `/build/assets/js` folder, and I reload the page, the SVGs finally disappear.

I'm not sure this is a bug, but I'd like to report it, and in case it's not a bug I'd like to understand which is the right way to do what I'm trying to do.

For screenshots that show better the issue, see the discussion https://github.com/facebook/docusaurus/discussions/8819

### Reproducible demo

https://github.com/0xCaso/polkadot-spec

### Steps to reproduce

```bash
npm i
npm run build_no_kaitai
npm run serve
```
- Then open in a text editor, for example, `/build/chap-state/chap-state.html`
- Look for 'graphviz', which is the class of the removed SVGs, and you should find nothing.
- If you go to the related page in the local website, you'll still be able to visualise the deleted SVGs.
- Now, if you go to `/build/assets/js` and delete all the generated files there and then refresh the page, you'll see that the SVGs disappears.

### Expected behavior

The expected behaviour would be that the SVGs are not loaded if deleted from the HTML file.

### Actual behavior

The SVGs are still present in the rendering.

### Your environment

- Public source code: https://github.com/0xCaso/polkadot-spec
- Docusaurus version used: 2.4.0
- Environment name and version (e.g. Chrome 89, Node.js 16.4): Brave v1.49.120, 
- Operating system and version (e.g. Ubuntu 20.04.2 LTS): MacBook Air M1, MacOS 13.2.1 (22D68)

### Self-service

- [ ] I'd be willing to fix this bug myself.

## UPDATE
I updated the code. The issue remains, but I deleted the example, and instead of deleting the SVGs, now the plugin actually does the edits that I want to do. You can still notice the issue: edits happen ONLY IF I clean the `/build/assets/js` directory.

The problem actually applies for both the custom plugins I made.

## UPDATE 2
Probably the one or more of the generated `js` files is because of the remark or rehype plugins (`math`, `katex`), or the default ones. I think the only way to solve this is establishing the plugins execution order, but as I've read it's not possible: docusaurus calls them in parallel.
I could treat the plugin I wrote as a standalone script, called after the build (and not as a plugin that runs in `postBuild`), but I think this won't solve anything as there will always be those `js` files that "hide" the edits.

**Comment by slorber** [comment]
@0xCaso sorry but I'm going to close this. We can pursue the existing discussion on the already existing discussion to figure out if there is a bug or not. A bug report requires a clear minimal repro of the problem, not throwing away a large repro and let me do the work to figure out what you even try to say. 

If you want to open a bug report, explain things clearly, and provide me a but report with only the relevant parts from a newly initialized Docusaurus site. There should be 0 mentions of anything related to Polkadot in the repro. Ideally, you'd add 2 commits: one for project init, and one for the modifications that led to the bug you encounter.

In other words, if you want me to help you, do the upfront work so that helping you is not a time-sink for me.
